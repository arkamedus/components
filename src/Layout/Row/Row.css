.row {
    width: 100%;
    display: grid;
    grid-template-columns: repeat(24, 1fr);
}

.row.gap {
    column-gap: var(--gap-sizing);
    row-gap: var(--gap-sizing);
}