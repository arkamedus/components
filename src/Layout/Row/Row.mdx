import { Controls, Meta, Primary, Stories } from "@storybook/blocks";
import meta from './Row.stories';

<Meta of={meta} />

# Row

The `Row` component is a foundational layout element designed to arrange content horizontally. It is intended primarily to contain `Column` components, which manage responsive behavior via props like `xs`, `sm`, `md`, `lg`, `xl`, and `xls`. Use `Row` to control spacing, alignment, and overall grid structure in your layout.

<Primary />
<Controls />

---

# Row Component Documentation

The `Row` component serves as the main container in a grid system. Its primary role is to organize and space out its child components—ideally, `Column` components—to create a consistent, responsive layout.

**Key Points:**
- **Layout Container:** Provides a horizontal structure for content.
- **Responsive Design:** Meant to host `Column` components that adjust based on viewport size.
- **Spacing & Alignment:** Manages gaps and alignment to maintain design consistency.

Listed below are additional variations of the component.

<Stories />
