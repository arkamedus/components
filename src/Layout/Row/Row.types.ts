import {CoreLayoutProps} from "../../Core/Core.types";

export interface RowProps extends CoreLayoutProps {

}