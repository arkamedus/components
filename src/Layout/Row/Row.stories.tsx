import React from "react";
import { Meta, StoryFn } from "@storybook/react";
import Row from "./Row";
import Col from "../Column/Column";
import DebugLayer from "../../Atom/DebugLayer/DebugLayer";
import Column from "../Column/Column";

const meta: Meta<typeof Row> = {
	title: "Design System/Layout/Row",
	argTypes: {
		pastel: {
			control: "boolean",
			description: "Toggle pastel background colors",
			defaultValue: false,
		},
	},
} as Meta;
export default meta;

const Template: StoryFn<{ pastel: boolean; gap: boolean }> = ({
	pastel,
	gap,
}) => (
	<Row className={pastel ? "pastel-container" : ""} gap={gap}>
		<Col xs={12}>
			<DebugLayer label="DebugLayer (Column xs=12)" />
		</Col>
		<Col xs={12}>
			<DebugLayer label="DebugLayer (Column xs=12)" />
		</Col>
	</Row>
);

export const Default = Template.bind({});
Default.args = { pastel: false, gap: false };

export const withColumnsGap = Template.bind({});
withColumnsGap.args = { pastel: true, gap: true };

export const withResponsiveColumnsGap = () => (
	<Row gap>
		<Column xs={24}>
			<DebugLayer label={"DebugLayer (Column 1)"} />
		</Column>
		<Column xs={24} md={12}>
			<DebugLayer />
		</Column>
		<Column xs={24} md={12}>
			<DebugLayer />
		</Column>
	</Row>
);
