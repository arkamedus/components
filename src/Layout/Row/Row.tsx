import React from "react";

import {RowProps} from "./Row.types";

import "./Row.css";

const Row: React.FC<RowProps> = ({style, children, className, gap}) => {
    const classNames = [className, "row"];
    if (gap){classNames.push("gap")}
    return (
        <div data-testid="Row" className={classNames.join(" ")} style={style}>{children}</div>
    )
};

export default Row;

