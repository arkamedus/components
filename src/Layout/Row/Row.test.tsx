import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Row from "./Row";

describe("Row Component", () => {
  const renderComponent = (props = {}) =>
      render(
          <Row data-testid="Row" {...props}>
            <div>Item 1</div>
            <div>Item 2</div>
          </Row>
      );

  it("should render without errors", () => {
    const { getByTestId } = renderComponent();
    expect(getByTestId("Row")).toBeInTheDocument();
  });

  it("should apply default classes", () => {
    const { getByTestId } = renderComponent();
    expect(getByTestId("Row")).toHaveClass("row");
  });

  it("should render children correctly", () => {
    const { getByTestId } = renderComponent();
    expect(getByTestId("Row")).toHaveTextContent("Item 1");
    expect(getByTestId("Row")).toHaveTextContent("Item 2");
  });

  it("should apply gap class when gap prop is true", () => {
    const { getByTestId } = renderComponent({ gap: true });
    expect(getByTestId("Row")).toHaveClass("gap");
  });

  it("should match snapshot", () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });
});
