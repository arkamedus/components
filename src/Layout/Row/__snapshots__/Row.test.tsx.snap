// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Component should match snapshot 1`] = `
<div>
  <div
    class="oakd row"
    data-testid="Row"
  >
    <div>
      Item 1
    </div>
    <div>
      Item 2
    </div>
  </div>
</div>
`;
